# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/angle/checkout/src/common/third_party/base/anglebase/numerics/safe_math.h
# Commit: 3a91885d47e0
# Full Hash: 3a91885d47e0ef7fba9749562d52300eea67c651
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2019-11-11 21:52:52
# Regressor Bug: 1578576
# File Overlap Count: 4
# Description:
#   Bug 1578576 - Part 2: Update to ANGLE 3865 r=jgilbert
#   
#   Depends on D44561
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44579
# ==============================================================================

diff -r 77cee77c0307 -r 3a91885d47e0 gfx/angle/checkout/src/common/third_party/base/anglebase/numerics/safe_math.h
--- a/gfx/angle/checkout/src/common/third_party/base/anglebase/numerics/safe_math.h	Fri Nov 08 16:41:42 2019 +0000
+++ b/gfx/angle/checkout/src/common/third_party/base/anglebase/numerics/safe_math.h	Fri Nov 08 17:06:00 2019 +0000
@@ -89,6 +89,14 @@
     // IsValid() is the public API to test if a CheckedNumeric is currently valid.
     bool IsValid() const { return validity() == RANGE_VALID; }
 
+    // AssignIfValid(Dst) - Assigns the underlying value if it is currently valid and is within the
+    // range supported by the destination type. Returns true if successful and false otherwise.
+    template <typename Dst>
+    constexpr bool AssignIfValid(Dst *result) const
+    {
+        return IsValid() ? ((*result = static_cast<Dst>(state_.value())), true) : false;
+    }
+
     // ValueOrDie() The primary accessor for the underlying value. If the current
     // state is not valid it will CHECK and crash.
     T ValueOrDie() const